### PR TITLE
Add bilateral_slice op

### PR DESCRIPTION
**Pull request by LielinJiang**
### PR types

New features 
### PR changes

OPs
### Describe

Add bilateral_slice op

1. 竞品为custom op，为支持vis同学线上业务
2. 竞品没有实现cpu kernel，并且只支持float32. 目前该op也只支持gpu。

### CODE REVIEW

File: paddle/fluid/operators/bilateral_slice_op.cu [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+  float abx = diff_abs(x);
+  if (abx > 1.0f) {
+    return 0.0f;
+  } else {
+    return d_diff_abs(x);
+  }
+}
+
+template <typename T>
+__global__ void bilateral_slice_cuda_forward_kernel(
```
Comment (qingqing01) [inline paddle/fluid/operators/bilateral_slice_op.cu]: Google C++ code style, function name: https://google.github.io/styleguide/cppguide.html#Function_Names

**Review by qingqing01 — COMMENTED** [review]
LGTM

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -12937,6 +12938,67 @@ def grid_sampler(x, grid, name=None):
...
 
 
+def bilateral_slice(x, guide, grid, has_offset, name=None):
+    """
+    :alias_main: paddle.nn.functional.bilateral_slice
+	:alias: paddle.nn.functional.bilateral_slice,paddle.nn.functional.vision.bilateral_slice
+	:old_api: paddle.fluid.layers.bilateral_slice
+
+    This operation slices input in in the location defined by guide, to produce output.
+    For more information, please refer to `Deep Bilateral Learning for Real-Time Image Enhancement <https://groups.csail.mit.edu/graphics/hdrnet/data/hdrnet.pdf>`_
```
Comment (SunGaofeng) [inline python/paddle/fluid/layers/nn.py]: This operation implements bilateral slicing on the input according to the guide map. 
For more information of bilateral slicing, please refer to `Deep Bilateral Learning for Real-Time Image Enhancement <https://groups.csail.mit.edu/graphics/hdrnet/data/hdrnet.pdf>`_

**Review by qingqing01 — DISMISSED** [review]

**Review by juncaipeng — DISMISSED** [review]
LGTM for changing op_accuracy_white_list.py and op_threshold_white_list.py

**Review by Aurelius84 — DISMISSED** [review]
LGTM for kernel registeration. please implement `double` kernel in future PR as mentioned in [here](http://agroup.baidu.com/paddlepaddle/view/office/2348282)

**Review by lanxianghit — APPROVED** [review]

**Review by juncaipeng — APPROVED** [review]
LGTM for changing op_accuracy_white_list.py and op_threshold_white_list.py

**Review by kolinwei — APPROVED** [review]
改OP无GPU实现，skip掉gpu单测

**Review by TCChenlong — APPROVED** [review]
LGTM

**Pull request merged**